### ISSUE TITLE
[BUG]: gemini 插件微调时，疑似不能正确加载模型权重

### ISSUE THREAD

**Issue by hmzo** [hpcaitech/ColossalAI#5034]
### 🐛 Describe the bug

# 环境
```
------------ Environment ------------
Colossal-AI version: 0.3.4
PyTorch version: 2.0.1
System CUDA version: 11.7
CUDA version required by PyTorch: 11.7
```

# BUG 细节
微调代码修改自：https://github.com/hpcaitech/ColossalAI/blob/main/examples/language/llama2/finetune.py

加载模型：https://github.com/hpcaitech/ColossalAI/blob/main/examples/language/llama2/finetune.py#L237

除了plugin的类型其余变量都保持一致，发现zero2时，loss的表现正常，而使用gemini时，更像是从一个随机初始化的weight进行优化

zero2，loss 正常从比较低的水平开始下降:
![image](https://github.com/hpcaitech/ColossalAI/assets/26140960/5ce96b4c-b85b-432e-8ed5-2dc58c7982ed)

gemini，loss 从特别高的水平下降：
![image](https://github.com/hpcaitech/ColossalAI/assets/26140960/e14c9afb-a112-44fd-a3ef-ca721d1eefa4)

**Comment by flybird11111** [comment]
感谢你的反馈，我们会尽快看一下这个问题。

**Comment by luckyyangrun** [comment]
When fine-tuning ChatGLM2 / Llama2, I also observed a similar phenomenon, under PyTorch version: 1.13.1. It seems that this phenomenon occurs when using Colossal-AI version >= 0.3.3.
